src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -194,6 +201,129 @@ public async Task ExecuteAsync(string stackName, bool saveCdkProject)
+        /// </summary>
+        /// <param name="recommendation">The selected recommendation settings used for deployment <see cref="Recommendation"/></param>
+        /// <param name="deploymentBundleDefinition">The container for the deployment bundle used by an application. <see cref="DeploymentBundleDefinition"/></param>
+        /// <param name="userDeploymentSettings">The deserialized object from the user provided config file. <see cref="UserDeploymentSettings"></param>
+        private async Task ConfigureDeployment(Recommendation recommendation, DeploymentBundleDefinition deploymentBundleDefinition, UserDeploymentSettings userDeploymentSettings)
+        {
+            foreach (var entry in userDeploymentSettings.OptionSettingItems)
+            {
+                var optionSettingJsonPath = entry.Key;
+                var optionSettingValue = entry.Value;
+
+                var isPartOfDeploymentBundle = true;
+                var optionSetting = deploymentBundleDefinition.Parameters.FirstOrDefault(x => x.Id.Equals(optionSettingJsonPath));
+                if (optionSetting == null)
+                {
+                    optionSetting = recommendation.GetOptionSetting(optionSettingJsonPath);
+                    isPartOfDeploymentBundle = false;
+                }
+                    
+                if (!recommendation.IsExistingCloudApplication || optionSetting.Updatable)
+                {
+                    object settingValue;
+                    try
+                    {
+                        switch (optionSetting.Type)
+                        {
+                            case OptionSettingValueType.String:
+                                settingValue = optionSettingValue;
+                                break;
+                            case OptionSettingValueType.Int:
+                                settingValue = int.Parse(optionSettingValue);
+                                break;
+                            case OptionSettingValueType.Bool:
+                                settingValue = bool.Parse(optionSettingValue);
+                                break;
+                            default:
+                                throw new InvalidOverrideValueException($"Invalid value {optionSettingValue} for option setting item {optionSettingJsonPath}");
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        throw new InvalidOverrideValueException($"Invalid value {optionSettingValue} for option setting item {optionSettingJsonPath}");
+                    }
+                    
+                    optionSetting.SetValueOverride(settingValue);
+
+                    if (isPartOfDeploymentBundle)
+                        SetDeploymentBundleOptionSetting(recommendation, optionSetting.Id, settingValue);
+
+                    if (optionSetting.Id.Equals("EC2KeyPair"))

```

Comment:
Removed the logic to create a _**new**_ EC2 Key Pair using the config file. If a Key Pair name is specified, it will be set as an OptionSettingValue. However, during deployment if found that that key pair does not already exist, then the deployment will fail either at the CDK level or the Cloud Formation level.